File: java/library/src/main/java/com/v2soft/styxlib/server/tcp/TCPClientDetails.java
package com.v2soft.styxlib.server.tcp;

import com.v2soft.styxlib.server.ClientDetails;
import com.v2soft.styxlib.server.IChannelDriver;

import java.io.IOException;
import java.nio.channels.SocketChannel;

/**
 * Created by V.Shcryabets on 5/20/14.
 *
 * @author V.Shcryabets (vshcryabets@gmail.com)
 */
public class TCPClientDetails extends ClientDetails<SocketChannel> {

    public TCPClientDetails(SocketChannel channel, IChannelDriver driver, int iounit, int id) {
        super(channel, driver, iounit, id);
    }

    @Override
    public String toString() {
        try {
            return String.format("%s:%d", mChannel.getRemoteAddress().toString(), mClientId);
        } catch (IOException e) {
            return super.toString();
        }
    }

    public void disconnect() throws IOException {
        // TODO something wrong, close should in same place where we have opened it.
        mChannel.close();
        mChannel = null;
    }
}

File: native/V2StyxLib/src/server/tcp/TCPClientDetails.cpp
/*
 * TCPClientState.cpp
 *
 *  Created on: May 2, 2018
 *      Author: vschryabets@gmail.com
 */

#include "server/tcp/TCPClientDetails.h"
#include <sys/socket.h>
#include "io/StyxDataReader.h"

TCPClientDetails::TCPClientDetails(Socket socket, IChannelDriver* driver, size_t iounit, int id) 
	: ClientDetails(driver, iounit, id), mChannel(socket) {
	if (socket == INVALID_SOCKET) {
		throw new StyxException("Socket is not ready");
	}
}

TCPClientDetails::~TCPClientDetails() {
}

Socket TCPClientDetails::getChannel() {
	return mChannel;
}

void TCPClientDetails::disconnect() throw(StyxException) {
#warning TODO something wrong, close should in same place where we have opened it.
	::shutdown(mChannel, SHUT_RDWR);
	mChannel = INVALID_SOCKET;
}

